### PR TITLE
fix: matching logic in autocomplete

### PR DESCRIPTION
**Pull request by cleanerzkp**
Closes
## Checklist

Fix autocomplete functionality and improve filtering logic
- Add exactMatchMatcher computed property to handle exact matches
- Update applySuggestion function to handle exact matches
- Modify watch on search to update selectedMatcher
- Emit 'update:keyword' event to update parent component

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Changelog
 * :feature:`8101` Failed transactions will now be displayed as failed and not just show "burned gas" event only.
 * :feature:`-` Transactions claiming SAFE tokens from vesting will now be properly decoded.
 * :feature:`-` Transactions locking, unlocking and withdrawing SAFE tokens will now be properly decoded. Also any locked SAFE tokens will be automatically detected and their balance counted.  
+* :bug:`` Improve autocomplete and filtering logic in FilterDropdown component.
```
Comment (kelsos) [inline docs/changelog.rst:19]:
```suggestion
* :bug:`-` Improve autocomplete and filtering logic in FilterDropdown component.
```

When we don't have an issue number you should add the dash in between.

**Comment by kelsos** [comment]
So the fix for `tx_hash` seems to work fine, but we probably should add a hint for assets since now it appears weird before you start typing.

I also get the following warning that we need to figure out.
```
[Vue warn] Write operation failed: computed value is readonly
```

Also since you are already touching can you re-order the components based on the [docs](https://docs.rotki.com/contribution-guides/vue-typescript.html#setup-script-order)

also a small note, in the project we use [vueuse](https://vueuse.org/) and auto-imports for [get](https://vueuse.org/shared/get/#get) and [set](https://vueuse.org/shared/set/#set) that we use instead of accessing `ref.value directly`

File: docs/changelog.rst [hunk 2]
```diff
@@ -16,11 +17,7 @@ Changelog
 * :feature:`4457` Rotki will show links to CoinGecko and CryptoCompare on the asset detail page if they exist.
 * :feature:`8101` Failed transactions will now be displayed as failed and not just show "burned gas" event only.
 * :feature:`-` Transactions claiming SAFE tokens from vesting will now be properly decoded.
-* :feature:`-` Transactions locking, unlocking and withdrawing SAFE tokens will now be properly decoded. Also any locked SAFE tokens will be automatically detected and their balance counted.
-* :bug:`-` rotki will now decode the swaps done on velodrome v2 in the right order.
-* :bug:`-` ZKSync Era tokens will now have prices queried properly by defillama.
-* :bug:`-` rotki will now query TheGraph delegations only for the addresses that interacted with the protocol.
-* :bug:`8452` Fix Monerium integration after the v2 contracts upgrade.
```
Comment (lukicenturi) [inline docs/changelog.rst:23]: Revert this

File: docs/changelog.rst [hunk 3]
```diff
@@ -16,11 +17,7 @@ Changelog
...
+* :feature:`-` Transactions locking, unlocking and withdrawing SAFE tokens will now be properly decoded. Also any locked SAFE tokens will be automatically detected and their balance counted.
```
Comment (lukicenturi) [inline docs/changelog.rst:20]: Revert this, just keep from develop

File: frontend/app/src/components/table-filter/FilterDropdown.vue [hunk 4]
```diff
@@ -1,4 +1,6 @@
 <script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+import { get, set } from '@vueuse/core';
```
Comment (lukicenturi) [inline frontend/app/src/components/table-filter/FilterDropdown.vue:3]: We have auto import mechanism, so these things should be able to removed

File: frontend/app/src/locales/fr.json [hunk 5]
```diff
@@ -4028,7 +4028,8 @@
     },
     "hint": {
       "description": "* tapez le filtre puis appuyez sur entrée, par exemple.",
-      "example": "start = 26/05/2022"
+      "example": "start = 26/05/2022",
+      "edit_note": "Double-click a filter to edit"
```
Comment (lukicenturi) [inline frontend/app/src/locales/fr.json:4032]: Nitpick, if you're not gonna translate this in other languages, just leave it in other locale files, and just fill the `en.json`

File: frontend/app/src/locales/en.json [hunk 6]
```diff
@@ -4307,7 +4307,9 @@
     },
     "hint": {
       "description": "* type the filter and then press enter e.g.",
-      "example": "start = 26/05/2022"
+      "example": "start = 26/05/2022",
+      "edit_note": "Double-click a filter to edit"
```
Comment (lukicenturi) [inline frontend/app/src/locales/en.json:4311]: Okay as I said this is not triggered by Double-click, but only single click, so:
```
Click applied filter to edit the value
```

File: frontend/app/src/components/table-filter/FilterDropdown.vue [hunk 7]
```diff
@@ -204,12 +253,12 @@ const highlightedTextClasses = 'text-subtitle-2 text-rui-text-secondary';
     </div>
     <div
       v-else-if="keyword && matchers.length === 0"
-      class="pb-2"
+      class="pb-0"
```
Comment (lukicenturi) [inline frontend/app/src/components/table-filter/FilterDropdown.vue:256]: just remove it and you won't have any padding (?)

File: frontend/app/src/locales/fr.json (hunk 5 above)
Comment (cleanerzkp) [reply frontend/app/src/locales/fr.json:4032]: I saw they had some English in them hence

File: frontend/app/src/components/table-filter/FilterDropdown.vue [hunk 8]
```diff
@@ -227,8 +272,11 @@ const highlightedTextClasses = 'text-subtitle-2 text-rui-text-secondary';
           :active="index === selectedSuggestion"
           :matcher="matcher"
           :class="{ highlightedMatcher: index === selectedSuggestion }"
-          @click="click($event)"
-        />
+          @click="click(matcher)"
+          @keydown.enter="handleEnter()"
+        >
+          <span class="font-bold text-rui-primary-main">{{ matcher.key }}:</span> {{ t(`table_filter.filter.${matcher.key}`) }}
```
Comment (lukicenturi) [inline frontend/app/src/components/table-filter/FilterDropdown.vue:278]: Does it do anything to you? Because FilterEntry doesn't even have a slot.

**Comment by lukicenturi** [comment]
> So the fix for `tx_hash` seems to work fine, but we probably should add a hint for assets since now it appears weird before you start typing.
> 
> I also get the following warning that we need to figure out.
> 
> ```
> [Vue warn] Write operation failed: computed value is readonly
> ```
> 
> Also since you are already touching can you re-order the components based on the [docs](https://docs.rotki.com/contribution-guides/vue-typescript.html#setup-script-order)
> 
> also a small note, in the project we use [vueuse](https://vueuse.org/) and auto-imports for [get](https://vueuse.org/shared/get/#get) and [set](https://vueuse.org/shared/set/#set) that we use instead of accessing `ref.value directly`

File: frontend/app/src/components/table-filter/FilterDropdown.vue [hunk 9]
```diff
@@ -136,11 +144,47 @@ watch(selectedSuggestion, async () => {
 });
 
 const highlightedTextClasses = 'text-subtitle-2 text-rui-text-secondary';
+
+function handleEnter() {
```
Comment (lukicenturi) [inline frontend/app/src/components/table-filter/FilterDropdown.vue:148]: what is this `handleEnter` does?

**Comment by lukicenturi** [comment]
image
weird UI if asset is selected

**Comment by lukicenturi** [comment]
image
hide this when no filter is selected

**Comment by lukicenturi** [comment]
This should not be here at this state

**Comment by lukicenturi** [comment]
the exclude one doesn't work: 

This is the develop
image

Your PR

**Comment by lukicenturi** [comment]
When I hit enter on "protocol: eas", it fill search input with "location="
Uploading Screen Recording 2024-09-05 at 15.51.36.mov…

File: frontend/app/src/components/table-filter/FilterDropdown.vue (hunk 9 above)
Comment (lukicenturi) [reply frontend/app/src/components/table-filter/FilterDropdown.vue:148]: I'm not sure where is this called, I put console.log here, it never gets called, by pressing enter while any button is highlighted

**Comment by lukicenturi** [comment]
In general I think you overcomplicate thing, especially for the issue when we set exact matcher for keys that have empty suggestions.
I was thinking the approach would as simple as, dont set it as "selectedMatcher" until we have any symbol :, !=, = set

Also, I see that you show that exactMatcher with the template that supposed to be used to show the suggestion value.

**Review by cleanerzkp — COMMENTED** [review]
LGTM!

**Review by lukicenturi — APPROVED** [review]

**Pull request merged**